### ISSUE TITLE
Wrong PATH for downloaded ruby for heroku-22 and above?

### ISSUE THREAD

**Issue by tkrevh** [heroku/heroku-buildpack-nginx#127]
Hello!

I'm trying to upgrade to heroku-22 and I'm running into issue that erb command needed by start-nginx script is not found.

Could it be, that the following line:
https://github.com/heroku/heroku-buildpack-nginx/blob/201ffe12f0d7ef2e8676814cdcec100cc87e10d7/bin/compile#L50

should actually reference `${BUILD_DIR}` instead of `${HOME}` ?

Like this:
`echo "export PATH=\"\${BUILD_DIR}/${vendored_ruby_dir}/bin:\${PATH}\"" > "${BUILD_DIR}/.profile.d/ruby.sh"`

If you look at line 40, the folder is created like this:
`mkdir -p "${BUILD_DIR}/${vendored_ruby_dir}"`

But when setting the path, another variable is used.

This works fine on heroku-20 and earlier versions as it seems they came with preinstalled Ruby.

**Comment by dzuelke** [comment]
No, `${HOME}` is correct - that `.profile.d` script is evaluated at dyno startup.

I can't reproduce this - Ruby gets installed, and the path set correctly, for my simple test case.

Are you using this in combination with another buildpack by any chance?

**Comment by dzuelke** [comment]
@tkrevh please provide output from the build, as well as logs from app startup. If you're using a custom Nginx config, please share that if you can. Also, a `heroku run bash` where you do `echo $PATH` and `which erb` would be helpful. Plus, like I already mentioned, any info on additional buildpacks you're using.

**Comment by chap** [comment]
[heroku-22+ no longer includes ruby](https://devcenter.heroku.com/articles/heroku-22-stack)

> The stack no longer includes a system Ruby installation. This will not affect the vast majority of users, since Ruby apps will use the Ruby installation provided by the Ruby buildpack.

**Comment by dzuelke** [comment]
That's not the issue though, @chap - the buildpack installs Ruby for this exact reason. Must be something else.

**Comment by dzuelke** [comment]
That would be very useful, @tkrevh. We're still unable to reproduce this.

**Comment by dzuelke** [comment]
Any news here, @tkrevh? Can you reproduce this easily on a new app?

**Comment by tkrevh** [comment]
I'm sorry, we just had a major release and our focus is elsewhere. I will revisit this later in the week.
Sorry and thank you for your patience!

**Comment by dzuelke** [comment]
All good, I know how it is sometimes, @tkrevh :)

**Comment by tkrevh** [comment]
@dzuelke thank you for your patience. I got back to this... just to realize what a stupid mistake I made.

I overlooked the fact, that we were using a fork of this buildpack.
I've merged the new compile script and added the missing start-nginx-static script to our forked buildpack repository,
and it's working.

Totally my fault, so sorry!

The only reason we're keeping our own fork of the buildpack is the correct support for SIGTERM. 
There is an open PR for this (and we've been using this fix in production for over 3 years so it's proven):
https://github.com/heroku/heroku-buildpack-nginx/pull/88

Any chance we could get it merged?

**Comment by dzuelke** [comment]
Ah, that explains a lot! :)

We'll take a look at #88 - the PHP buildpack does something similar for its Nginx build (and HTTPD, and PHP-FPM).

**Comment by tkrevh** [comment]
Thank you, appreciate it 🙏
